ports/nrf/common-hal/digitalio/DigitalInOut.c

```diff
@@ -24,146 +24,148 @@
     self->open_drain = (drive_mode == DRIVE_MODE_OPEN_DRAIN);
 
-    hal_gpio_cfg_pin(self->pin->port, pin, HAL_GPIO_MODE_OUTPUT, HAL_GPIO_PULL_DISABLED);
+    nrf_gpio_cfg_input(pin, NRF_GPIO_PIN_NOPULL);
+
     common_hal_digitalio_digitalinout_set_value(self, value);
 }
 
 digitalio_direction_t common_hal_digitalio_digitalinout_get_direction(
-        digitalio_digitalinout_obj_t* self) {
-    return self->output? DIRECTION_OUTPUT : DIRECTION_INPUT;
+        digitalio_digitalinout_obj_t *self) {
+    const uint32_t pin = NRF_GPIO_PIN_MAP(self->pin->port, self->pin->pin);
+
+    return (nrf_gpio_pin_dir_get(pin) == NRF_GPIO_PIN_DIR_OUTPUT) ? DIRECTION_OUTPUT : DIRECTION_INPUT;
 }
 
 void common_hal_digitalio_digitalinout_set_value(
-        digitalio_digitalinout_obj_t* self, bool value) {
-    if (value) {
-        if (self->open_drain) {
-            hal_gpio_dir_set(self->pin->port, self->pin->pin, HAL_GPIO_MODE_INPUT);
-        } else {
-            hal_gpio_pin_set(self->pin->port, self->pin->pin);
-            hal_gpio_dir_set(self->pin->port, self->pin->pin, HAL_GPIO_MODE_OUTPUT);
-        }
+        digitalio_digitalinout_obj_t *self, bool value) {
+    const uint32_t pin = NRF_GPIO_PIN_MAP(self->pin->port, self->pin->pin);
+
+    if (value && self->open_drain) {
+        nrf_gpio_pin_dir_set(pin, NRF_GPIO_PIN_DIR_INPUT);
     } else {
-        hal_gpio_pin_clear(self->pin->port, self->pin->pin);
-        hal_gpio_dir_set(self->pin->port, self->pin->pin, HAL_GPIO_MODE_OUTPUT);
+        nrf_gpio_pin_dir_set(pin, NRF_GPIO_PIN_DIR_OUTPUT);
+        nrf_gpio_pin_write(pin, value);
     }
 }
 
 bool common_hal_digitalio_digitalinout_get_value(
-        digitalio_digitalinout_obj_t* self) {
-    const uint8_t pin = self->pin->pin;
-    if (!self->output) {
-        return hal_gpio_pin_read(self->pin);
-    } else {
-        if (self->open_drain && hal_gpio_dir_get(self->pin->port, self->pin->pin) == HAL_GPIO_MODE_INPUT) {
+        digitalio_digitalinout_obj_t *self) {
+    const uint32_t pin = NRF_GPIO_PIN_MAP(self->pin->port, self->pin->pin);
+    const nrf_gpio_pin_dir_t dir = nrf_gpio_pin_dir_get(pin);
+
+    if (dir == NRF_GPIO_PIN_DIR_INPUT) {
+        if (self->open_drain)
             return true;
-        } else {
-            return (GPIO_BASE(self->pin->port)->OUT >> pin) & 1;
-        }
+
+        return nrf_gpio_pin_read(pin);
     }
+
+    return nrf_gpio_pin_out_read(pin);
 }
 
 void common_hal_digitalio_digitalinout_set_drive_mode(
-        digitalio_digitalinout_obj_t* self,
+        digitalio_digitalinout_obj_t *self,
         digitalio_drive_mode_t drive_mode) {
-    bool value = common_hal_digitalio_digitalinout_get_value(self);
+    const bool value = common_hal_digitalio_digitalinout_get_value(self);
     self->open_drain = drive_mode == DRIVE_MODE_OPEN_DRAIN;
+
     // True is implemented differently between modes so reset the value to make
     // sure its correct for the new mode.
-    if (value) {
+    if (value)
```

Comment:
I know this is already approved, but I would ask you not to remove braces around single statements in control structures. Please always use braces in control structures, except perhaps in the most extreme cases (like very stylized single-line code). I've had to fix several bugs over the years that were caused by edits to missing braces.

Reply:
Sorry, I'll try to remember, at this point it's muscle memory :sweat_smile: